### PR TITLE
Update snowflake driver to latest version

### LINKED ISSUE #25849: Update Snowflake Drivers
> This announcement is to let you know that all partner and customer Snowflake accounts hosted on Google Cloud must update their client drivers to at least the minimum versions listed below by November 1, 2023. These new minimum versions, and all subsequent versions, provide critical fixes needed to continue accessing Snowflake on Google Cloud. Snowflake will therefore no longer support versions older than the new minimum versions for Google Cloud accounts. 
>
> Please note that the deprecation does not impact AWS or Azure partners or customers, who can continue to use driver versions released within the last two years as per the Snowflake Support policy. 
>
> WHY ARE CLIENT UPGRADES REQUIRED FOR GOOGLE CLOUD CUSTOMERS?
Historically, presigned URL API requests to Snowflake GCP buckets only required the payload to be signed. However Google now requires both the request headers and payload to be signed. After the cutover date, once this requirement is enforced, GET and PUT requests to the stage from clients below the minimum version will fail with HTTP Error 400.
>
> To avoid these failures, Snowflake has decided in consultation with Google to switch all drivers from using presigned URL authentication to using GCS downscoped token authentication. 
>
> ACTION TO TAKE:
In response to Google’s changes, Snowflake has released a new set of drivers. All customers with their Snowflake account hosted on Google Cloud must update their client drivers to these minimum versions or later listed below by November 1, 2023.
>
> Partners should also upgrade drivers for their SaaS environments, and should test, and bundle, or recommend at least the minimum supported driver versions for GCP accounts for their customer managed environments. Note that all customers with Snowflake accounts hosted on GCP will receive a communication about this change, so they may reach out to you to ensure that you’re using a supported client driver.
>
> Snowflake will ensure that the older drivers continue to work for GCP accounts until Nov 1, 2023. They will be supported as per the Snowflake Client and Version Support policy for this duration.
>
> After the cutover date, the GET and PUT requests from client drivers below the minimum version will fail with HTTP Error 400. Please see this [KB article for more details](https://go.snowflake.net/MjUyLVJGTy0yMjcAAAGLQ4R7SqWTersInzGkOuD0qClFKQ7JQEQ5NsXaOCdYhhYNNFwGATMBz9aiUshz45Cv0qJ8oQ0=). Please see [this article](https://go.snowflake.net/MjUyLVJGTy0yMjcAAAGLQ4R7SxKvfbmKBo89Kqi9m_53O-eSw-vEKu0yT7zMikoFj0PhWYFkPX_8w_fh2Q03Tc28Dds=) for more information on how to check your current client driver versions in Snowflake.

### PR DESCRIPTION
**Pull request by evantahler**
Closes https://github.com/airbytehq/airbyte/issues/25849

### CODE REVIEW

File: airbyte-integrations/connectors/source-snowflake/build.gradle [hunk 1]
```diff
@@ -17,7 +17,7 @@ dependencies {
     implementation project(':airbyte-integrations:connectors:source-relational-db')
     implementation libs.airbyte.protocol
     implementation files(project(':airbyte-integrations:bases:base-java').airbyteDocker.outputs)
-    implementation group: 'net.snowflake', name: 'snowflake-jdbc', version: '3.13.22'
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-snowflake/build.gradle:20]: The source and destination driver versions didn't match before this PR...
